streampipes-processors-transformation-jvm/src/main/java/org/apache/streampipes/processors/transformation/jvm/processor/booloperator/logical/BooleanOperatorProcessor.java

```diff
@@ -0,0 +1,129 @@
+import org.apache.streampipes.sdk.helpers.EpRequirements;
+import org.apache.streampipes.sdk.helpers.Labels;
+import org.apache.streampipes.sdk.helpers.Locales;
+import org.apache.streampipes.sdk.helpers.Options;
+import org.apache.streampipes.sdk.helpers.OutputStrategies;
+import org.apache.streampipes.sdk.utils.Assets;
+import org.apache.streampipes.sdk.utils.Datatypes;
+import org.apache.streampipes.wrapper.context.EventProcessorRuntimeContext;
+import org.apache.streampipes.wrapper.routing.SpOutputCollector;
+import org.apache.streampipes.wrapper.standalone.ProcessorParams;
+import org.apache.streampipes.wrapper.standalone.StreamPipesDataProcessor;
+
+import java.util.List;
+
+import static org.apache.streampipes.processors.transformation.jvm.processor.booloperator.logical.enums.BooleanOperatorType.NOT;
+
+public class BooleanOperatorProcessor extends StreamPipesDataProcessor {
+
+    private static final String BOOLEAN_PROCESSOR_OUT_KEY = "boolean-operations-result";
+    private static final String BOOLEAN_OPERATOR_TYPE = "operator";
+    private static final String PROPERTIES_LIST = "properties";
+    private BooleanOperationInputConfigs configs;
+
+    @Override
+    public DataProcessorDescription declareModel() {
+        return ProcessingElementBuilder.create("org.apache.streampipes.processors.transformation.jvm.processor.booloperator.logical")
+                .withAssets(Assets.DOCUMENTATION, Assets.ICON)
+                .withLocales(Locales.EN)
+                .category(DataProcessorType.ENRICH)
+                .requiredStream(
+                        StreamRequirementsBuilder
+                                .create()
+                                .requiredPropertyWithUnaryMapping(EpRequirements.booleanReq(),
+                                        Labels.withId(PROPERTIES_LIST), PropertyScope.NONE)
+                                .build())
+                .requiredSingleValueSelection(Labels.withId(BOOLEAN_OPERATOR_TYPE), Options.from(
+                        BooleanOperatorType.AND.operator(),
+                        BooleanOperatorType.OR.operator(),
+                        BooleanOperatorType.NOT.operator(),
+                        BooleanOperatorType.XOR.operator(),
+                        BooleanOperatorType.X_NOR.operator(),
+                        BooleanOperatorType.NOR.operator()))
+                .outputStrategy(OutputStrategies.append(
+                        PrimitivePropertyBuilder.create(
+                                Datatypes.String, BOOLEAN_PROCESSOR_OUT_KEY)
+                                .build()))
+                .build();
+    }
+
+    @Override
+    public void onInvocation(ProcessorParams processorParams, SpOutputCollector spOutputCollector, EventProcessorRuntimeContext eventProcessorRuntimeContext) throws SpRuntimeException {
+        List<String> properties = processorParams.extractor().getUnaryMappingsFromCollection(PROPERTIES_LIST);
```

Comment:
```suggestion
        List<String> properties = processorParams.extractor().mappingPropertyValues((PROPERTIES_LIST);
```
due to change in `declareModel`